python/cuml/cuml/internals/base.pyx

```diff
@@ -988,3 +994,27 @@ class UniversalBase(Base):
         estimator.output_mem_type = MemoryType.host
 
         return estimator
+
+    def get_params(self, deep=True):
+        """
+        If accelerator is active, we return the params of the CPU estimator
+        being helf by the class, otherwise we just call the regular
+        get_params of the Base class.
+        """
```

Comment:
Triggered by Simon's comment: should this maybe be a comment instead of a docstring? It reads more like a comment for other devs, and less like documentation for users.

Below a suggestion with the `get_params` docstring from scikit-learn

```suggestion
        """
        Get parameters for this estimator.

        Parameters
        ----------
        deep : bool, default=True
            If True, will return the parameters for this estimator and
            contained subobjects that are estimators.

        Returns
        -------
        params : dict
            Parameter names mapped to their values.
        """
```

Reply:
`get_params` is part of the official API of a scikit-learn estimator just like `fit` or `predict`. So I think it deserves a good docstring, especially as it is free to get one

Reply:
You are 100% correct and I did not realize that we were referencing one of those public functions.